Added m2m for licence in `Assign licence` transition.

xliiv opened the pull request:
Before:
User could pick only one licence in `Assign licence` tranision.
Now:
User can pick many licences in `Assign licence` tranision.

mkurek commented on src/ralph/back_office/models.py:
```diff
@@ -229,10 +231,10 @@ def assign_user(cls, instances, request, **kwargs):
-            )
+            for pk in kwargs['licence']:
+                BaseObjectLicence.objects.get_or_create(
+                    base_object=instance, licence=Licence.objects.get(pk=pk)
```
why not `base_object=instance, licence_id=pk`?

mkurek commented on src/ralph/lib/transitions/models.py:
```diff
@@ -101,8 +101,11 @@ def _get_history_dict(data, instance, runned_funcs):
             except FieldDoesNotExist:
                 field = func.form_fields[k]['field']
-                if isinstance(field, forms.ChoiceField):
+                if type(field) == forms.ChoiceField:
```
i'd prefer not to use `type` comparison. you could swap these if-else to first check `if isinstance(field, forms.ModelMultipleChoiceField):`, then `elif isinstance(field, forms.ChoiceField):`

mkurek commented on src/ralph/lib/transitions/views.py:
```diff
@@ -62,10 +62,15 @@ def form_fields_from_actions(self):
                     )
+                    widget_options = {}
+                    autocomplete_name = options.get('autocomplete_field', None)
+                    if autocomplete_name in ['licence']:
```
http://www.nooooooooooooooo.com/
please add this as another parameter to transitions fields definition or deduct if from field type

mkurek commented on src/ralph/back_office/models.py:
```diff
@@ -221,18 +221,21 @@ def assign_user(cls, instances, request, **kwargs):
     @transition_action(
         form_fields={
             'licence': {
```
licences?

mkurek commented on src/ralph/back_office/tests/test_models.py:
```diff
@@ -296,6 +297,20 @@ def test_convert_to_data_center_asset(self):
+            instances=[asset], request=None, licence=licences,
+        )
+
+        self.assertEqual(
```
I'd use https://docs.python.org/3.4/library/unittest.html#unittest.TestCase.assertCountEqual here

mkurek commented on src/ralph/lib/transitions/views.py:
```diff
@@ -62,10 +62,12 @@ def form_fields_from_actions(self):
                         field=field,
                         admin_site=ralph_site,
                         request=self.request,
+                        **widget_options
```
why not just `**options.get('widget_options', {})`?

szok commented:
:ok:

mkurek commented:
:ok:

Commit:
Added m2m for licence in `Assign licence` transition.

Before:
User could pick only one licence in `Assign licence` tranision.
Now:
User can pick many licences in `Assign licence` tranision.
